libs/platform/experience/src/controllers/layout.controller.ts

```diff
@@ -37,9 +106,260 @@ export class LayoutController {
       .pipe(map((layoutStyles) => `${layoutStyles}${componentStyles}`));
   }
 
+  /**
+   * @deprecated since 1.2 will be removed.
+   */
+  collectStyles(
+    layoutProperties: (keyof LayoutProperties)[],
+    rules: StyleRuleSet[] = [],
+    uid?: string
+  ): string {
+    let styles = '';
+
+    if (!this.hasLayout(rules, layoutProperties)) {
+      styles += ':host {display: contents;}\n';
+    }
+
+    styles += this.layoutBuilder.createStylesFromOptions(rules, uid);
+
+    return styles;
+  }
+
+  getRender(config: LayoutRenderParams): Observable<TemplateResult> {
+    const { screen, attrs, place, data } = config;
+
+    return this.getLayoutOptions(attrs, data.options?.rules, screen).pipe(
+      switchMap((layoutOptions) =>
+        from(Object.entries(layoutOptions)).pipe(
+          concatMap(([prop, value]) => {
+            if (!value) return of(null);
+
+            const token = prop === 'layout' ? (value as string) : prop;
+            const type =
+              prop === 'layout'
+                ? LayoutPluginType.Layout
+                : LayoutPluginType.Property;
+
+            return this.layoutService.getRender({
+              type,
+              token,
+              data: {
+                ...data,
+                options: layoutOptions,
+              },
+            });
+          }),
+          reduce(
+            (acc, curr) => (!curr?.[place] ? acc : html`${acc}${curr[place]}`),
+            html``
+          )
+        )
+      )
+    );
+  }
+
+  protected getLayoutOptions(
+    attrs: string[],
+    styles: StyleRuleSet[] = [],
+    screen?: string
+  ): Observable<LayoutProperties> {
+    const host = this.host;
+
+    const hostProperties = ['layout', ...attrs].reduce((acc, attr) => {
+      const value = host[attr as keyof typeof host];
+
+      if (attr === 'layout' && value) {
+        return { ...acc, layout: value };
+      }
+
+      if (host.hasAttribute(attr)) {
+        const prop = attr
+          .replace('layout-', '')
+          .replace(/-([a-z])/g, (g) => g[1].toUpperCase());
```

Comment:
Can we use just `toUpperCase` ?

```suggestion
        const prop = attr.replace('layout-', '').toUpperCase();
```

Reply:
we transform it to camelCase